### PR TITLE
Fix missing inline keyword which leads to duplicated symbol during linking

### PR DESCRIPTION
**Pull request by ss-abramchuk**

### CODE REVIEW

**Comment by ordex** [comment]
Hi, what makes you think that the symbol is duplicate during linking?

File: openvpn/common/hexstr.hpp [hunk 1]
```diff
@@ -180,7 +180,7 @@ namespace openvpn {
 	dest.push_back((high<<4) + low);
       }
     if (i != len)
-      throw parse_hex_error(); // straggler char      
+      throw parse_hex_error(); // straggler char
```
Comment (ordex) [inline openvpn/common/hexstr.hpp]: code cleanups and style fixes should always go in their own commits and not merged with other changes.

**Comment by ss-abramchuk** [comment]
Hi @ordex, without `inline` keyword I got this error during building in Xcode
```
ld: 2 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Also readme Security section contains this paragraph:
```
OpenVPN 3 is a "header-only" library, therefore all free functions outside of classes should have the inline attribute.
```
So I guess `inline` attribute should present here.

File: openvpn/common/hexstr.hpp (hunk 1 above)
Comment (ss-abramchuk) [inline openvpn/common/hexstr.hpp]: Sorry for that, I guess Atom fixed this part for me and I haven't noticed that. Should I fix it?

**Comment by ordex** [comment]
Hi @ss-abramchuk , I hadn't seen any error/warning with gcc about that.
I guess it makes the decision on its own (?). What version of clang are you using?
Are you compiling with -std=c++14?

I see those are just standalone functions and not class methods, however I am not sure about what the c++14 standard says in this regards.

Maybe somebody else could chime in? @jamesyonan any opinion?

File: openvpn/common/hexstr.hpp (hunk 1 above)
Comment (ordex) [inline openvpn/common/hexstr.hpp]: remove that change or just create another commit with this change only. If at some point we have to revert a change due to a bug, we don't want to revert other unrelated changes.

**Comment by ss-abramchuk** [comment]
It compiles fine if the project has only one `#import <client/ovpncli.hpp>` statement. As soon as I add another `#import` of any OpenVPN header which has included `<openvpn/common/hexstr.hpp>` somewhere I get the mentioned above error. And as I see all functions except `render_hex_number` have `inline` attribute so I guess it was just missed.

As for version and standard I have the following:
```
ss-imac:~ ss.abramchuk$ clang --version
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
---
CLANG_CXX_LANGUAGE_STANDARD = gnu++14
CLANG_CXX_LIBRARY = libc++
```

I tried `-std=c++14` and got the same result.

**Comment by dsommers** [comment]
Uhm .... is your issue really resolved, @ss-abramchuk?  I see the pull req #5 reference, but this is GitHub being confused, as we use Jira/Bitbucket internally (this openvpn3 is just one of many projects we're working on).  And that OVPN3-211 ticket which got merged is something completely different.

**Comment by ss-abramchuk** [comment]
Actually, it is still not resolved. Due to multiple merging my fork looks a bit messy. I'm going to create a new pull request that focuses only on missing `inline` to keep history clean.

**Comment by schwabe** [comment]
@ss-abramchuk actually I have another pull request for a similar reason that will also fix this (Did that idependendtly of you). So no action needed on your part. And btw. inline is not needed for the template version since templates are always weak linked/kind of inline.
